test/ui/InitializationTest.ts

```diff
@@ -353,22 +356,66 @@ export function InitializationTest() {
       Simulate.removeJQuery();
     });
 
-    it('will trigger a query automatically by default', done => {
-      Initialization.initializeFramework(root, searchInterfaceOptions, () => {
-        return Initialization.initSearchInterface(root, searchInterfaceOptions);
-      }).then(() => {
-        expect(endpoint.search).toHaveBeenCalled();
-        done();
+    describe('with automatic first query', () => {
+      let spyOnQuery: jasmine.Spy;
+
+      const doInit = () => {
+        return Initialization.initializeFramework(root, searchInterfaceOptions, () => {
+          const initResult = Initialization.initSearchInterface(root, searchInterfaceOptions);
+          spyOnQuery = spyOn(get(root, QueryController) as QueryController, 'executeQuery');
+          return initResult;
+        });
+      };
+
+      beforeEach(() => {
+        searchInterfaceOptions['SearchInterface'].autoTriggerQuery = true;
       });
-    });
 
-    it('will not trigger a query automatically if specified', done => {
-      searchInterfaceOptions['SearchInterface'].autoTriggerQuery = false;
-      Initialization.initializeFramework(root, searchInterfaceOptions, () => {
-        return Initialization.initSearchInterface(root, searchInterfaceOptions);
-      }).then(() => {
-        expect(endpoint.search).not.toHaveBeenCalled();
-        done();
+      afterEach(() => {
+        spyOnQuery = null;
+      });
+
+      it('will trigger a query automatically by default', done => {
+        doInit().then(() => {
+          expect(spyOnQuery).toHaveBeenCalled();
+          done();
+        });
+      });
+
+      it('will not trigger a query automatically if specified', done => {
+        searchInterfaceOptions['SearchInterface'].autoTriggerQuery = false;
+        doInit().then(() => {
+          expect(spyOnQuery).not.toHaveBeenCalled();
+          done();
+        });
+      });
+
+      describe('when query with no keywords are not allowed', () => {
+        beforeEach(() => {
+          searchInterfaceOptions['SearchInterface'].allowNoKeywords = false;
+        });
+
+        it('will trigger a query automatically if the interface contains keywords', done => {
+          $$(root).on(InitializationEvents.afterInitialization, () => {
```

Comment:
Why is it necessary to add this event listener?

Reply:
Because I cannot call the `state` without having first initialized the interface. The method needs a working search interface.